### PR TITLE
chore: auth login and signup complete

### PR DESCRIPTION
**Pull request by Mohitkumar33**
Authentication complete
Login complete
signup complete

### CODE REVIEW

File: src/pages/signup/signup.css [hunk 1]
```diff
@@ -0,0 +1,92 @@
+/* :root {
```
Comment (deepak29-git) [inline src/pages/signup/signup.css:1]: remove commented code

File: src/pages/signup/signup.js [hunk 2]
```diff
@@ -0,0 +1,121 @@
+import { Link, useNavigate } from "react-router-dom";
+import { useState } from "react";
+import { useAuth } from "../../contexts/auth-context";
+import "./signup.css";
+import axios from "axios";
+useState;
+const Signup = () => {
+  const [firstName, setFirstName] = useState("");
```
Comment (deepak29-git) [inline src/pages/signup/signup.js:8]: suggestion: you can use object instead of four use state

**Review by deepak29-git — APPROVED** [review]

File: src/pages/signup/signup.js (hunk 2 above)
Comment (Mohitkumar33) [reply src/pages/signup/signup.js:8]: so you are saying that I can take one object and add all the states in that. Thanks for the suggestion

File: src/pages/login/login.js [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+  const [password, setPassword] = useState("");
+  const callUserLogin = async (emailId, passw) => {
+    try {
+      const { data } = await axios.post("/api/auth/login", {
+        email: emailId,
+        password: passw,
+      });
+      localStorage.setItem("token", data.encodedToken);
+      localStorage.setItem("userName", data.foundUser.firstName);
+      console.log(localStorage.token, localStorage.userName);
```
Comment (prathmeshjagtap) [inline src/pages/login/login.js:20]: remove console.log

**Review by prathmeshjagtap — APPROVED** [review]
else everything  is fine

File: src/App.js [hunk 4]
```diff
@@ -21,6 +22,8 @@ function App() {
         <Route path="/history" element={<History />} />
         <Route path="/watchlater" element={<WatchLater />} />
         <Route path="/liked" element={<Liked />} />
+        <Route path="/login" element={<Login />} />
+        <Route path="/signup" element={<Signup />} />
         <Route path="/mock" element={<MockAPI />} />
         {/* <Route path="*" element */}
       </Routes>
```
Comment (dikshadhote) [inline src/App.js]: Can make separate file for routing and then import here

File: src/pages/login/login.js [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+      setAuthState({
+        isAuth: localStorage.token ? true : false,
+        userInfo: localStorage.userName ? localStorage.userName : null,
+      });
+      navigate("/");
+    } catch (error) {
+      console.log(error);
```
Comment (dikshadhote) [inline src/pages/login/login.js]: Instead of console.log can use throw(error)

**Review by dikshadhote — APPROVED** [review]
Very well written code.
Can use object to put all login field names

File: src/pages/login/login.js (hunk 5 above)
Comment (Mohitkumar33) [reply src/pages/login/login.js]: Yes okk thanks

File: src/App.js (hunk 4 above)
Comment (Mohitkumar33) [reply src/App.js]: yes thanks

**Pull request merged**